Sources/Utility/Process.swift

```diff
@@ -179,7 +179,10 @@ public final class Process: ObjectIdentifierProtocol {
 
         // Set the attribute flags.
         var flags = POSIX_SPAWN_SETSIGMASK | POSIX_SPAWN_SETSIGDEF
-        flags |= POSIX_SPAWN_SETPGROUP
+        // To allow taking input from terminal, only set this flag when stdin is not a tty.
+        if !isatty(STDIN_FILENO) {
+           flags |= POSIX_SPAWN_SETPGROUP
+        }
```

Comment:
I wonder if there is a problem between this and our killing behavior, because when we kill the subprocess we try to kill its entire process group... if we didn't create a PGROUP, that won't work -- shouldn't that cause a test failure with this change?